### ISSUE TITLE
Error message while using trackPath function

### ISSUE THREAD

**Issue by Sylvain6259** [aharmer/pathtrackr#2]
Hi aharmer,
I am Sylvain , I currently work at UTMSI in Texas. I would like to track animal movements using your toolbox and R. 

I have downloaded your toolbox as well as installed FFmpeg. But I got some message errors from R, and I can't resolve the issues myself. Maybe you could help me. 
First, I have a video which is a .avi file, I am not sure whether or not I can use the `splitVideo `function. Does the format has to be mpeg? (related to FFmpeg?). I did try to convert the video into jpeg files using `splitVideo`, but without any success. 
The error message is :
Warning messages:
```
1: running command 'ffmpeg -loglevel panic -y -i D:/ptrack -vf scale=320:-1 D:/ptrack' had status 127 
2: running command 'ffmpeg -loglevel panic -y -i D:/ptrack -q:v 2 -vf fps=30 -b:v 2000 -bt 20M D:/ptrack/frame%06d.jpg' had status 127 
```
And the jpeg files are not created. 
My video is a .avi file 30 frames/s and about 1920x1552 pix (21Giga).
Maybe the problem comes from the size or the format of the video. 

So I have decided to pass by Matlab to create jpeg files (I am a beginner with R), so from the .avi, I have created jpeg files. 
Then, I have tried to use with `trackPath `function. 
And I can not. R does not stop saying :

`Error in .rasterObjectFromFile(x, band = band, objecttype = "RasterLayer",  : Cannot create a RasterLayer object from this file.`
I have tried to figure out where this error comes from. I installed _rgdal_ package as suggested by R, and then I installed _raster_ package too, without more success. 
It seems that R can not create rasters from the jpeg I have. 
I don't know what to do, maybe the fact that I have created my jpeg with another function that the one in the `pathtrackr `package is the source of the error, or the size of the jpeg
I use Windows vista 32b and  R version 3.3.2 (2016-10-31) with RStudio. All the packages are up to date.
Any idea? 
Cheers, 
Sylvain

**Comment by Sylvain6259** [comment]
Hi again, 
I have the feeling that the problem is caused by this code line in `trackPath`: 
`  plot(raster::raster(file.list[1], band = 2), col = gray.colors(256))
`
If I try the function `raster `itself , I can create a raster. if I try the function `raster `with `band = 2`, it does not work. `band` being the layer to use in a multiple layer files, I guess I have only one layer....

**Comment by aharmer** [comment]
Hi Sylvain,

Can you first just confirm the file path you are using in the splitVideo function? This should point directly to the video file itself, not just the directory where it is installed. Also, avi files should not be a problem.

**Comment by Sylvain6259** [comment]
Hi Aharmer, 
Thank for the answer. 
I tried different things : 
 first `splitVideo("D:/ptrack", 30, xpix = 320, ypix = -1)`
Where D:/ptrack is the folder where my video file "essai.avi" is located. So that was not theright thing as you mentionned but 
I also tried
`splitVideo("D:/ptrack/essai", 30, xpix = 320, ypix = -1)`
or
`splitVideo("D:/ptrack/essai.avi", 30, xpix = 320, ypix = -1)`

Al of them return the status 127 error:
Warning messages:
```
1: running command 'ffmpeg -loglevel panic -y -i D:/ptrack/essai.avi -vf scale=320:-1 D:/ptrack/essai_COMPRESSED.avi' had status 127 
2: running command 'ffmpeg -loglevel panic -y -i D:/ptrack/essai_COMPRESSED.avi -q:v 2 -vf fps=30 -b:v 2000 -bt 20M D:/ptrack/essai/frame%06d.jpg' had status 127

```

**Comment by bpomalle** [comment]
Hi there, 

Seems like a great R package! Unfortunately I'm have the same 2 error messages as Sylvain. I keep getting status 127 and it won't split my video. Any advice would be much appreciated. thanks,

**Comment by Sylvain6259** [comment]
Hi bpomall, 
I have never figured out how to solve the poblem.
Sylvain

**Comment by dbthomas** [comment]
Hi Sylvain6259 and bpomall,

To elaborate on the replies from aharmer above:
There are two required terms for the `splitVideo `function, `filepath `and `fps`. These need to be written as per below:
`splitVideo(filepath="D:/Projects/Aaron_spider/test.mov", fps=30)`
The `xpix `and `ypix `values are important if you want to specify a particular aspect ratio, but pathtrackR will just use default values so they are not required terms.
The required terms need to be written as `filepath = "..."` and `fps = "..."`. The example from Sylvain6259  from Feb 15 just lists the `fps `value (30) but does not tell `splitVideo `that this is the `fps `value, i.e. `fps=30`.

To troubleshoot the example from Sylvain6259 I would try:
`splitVideo("D:/ptrack/essai.avi", fps=30)`

**Comment by Sylvain6259** [comment]
Thank you dbthomas for your help. 
I tried several things, but it is weeks ago, so I don't remember whether I have tried what you have proposed. 
I will try it.
Thank you
Sylvain

**Comment by bpomalle** [comment]
Thank you. I figured out my issue - I didn't have ffmpeg fully installed before.

**Comment by Sylvain6259** [comment]
Hi again, 
I have just installed again R, Rstudio, Path track and ffmpeg. And now , when I try splitVideo("D:/ptrack/essai.avi", fps=30)
I got an error. I think it is about ffmpeg again.
Warning messages:
1: running command 'ffmpeg -loglevel panic -y -i D:/ptrack/essai.avi -vf scale=320:-1 D:/ptrack/essai_COMPRESSED.avi' had status 313 
2: running command 'ffmpeg -loglevel panic -y -i D:/ptrack/essai.avi-q:v 2 -vf fps=30 -b:v 2000 -bt 20M D:/ptrack/essai.avi/frame%06d.jpg' had status 313 

Furthermore windows says 
 "le point d'entrée GetNumaNodeProcessor MaskEx est introuvable"
 basically, "the entrance point for GetNumaNodeProcessorMaskEx is not found" that has something to do with KERNEL 32.dll

**Comment by dbthomas** [comment]
Hi Sylvain6259,
You are encountering compatibility issues between ffmpeg and your system, and not with pathtrackR per se. The splitVideo function in pathtrackR calls to ffmpeg to split the video into individual frames, and then the downstream functions in pathtrackR do the actual path tracking. I would suggest that you need a work around solution that let's you bypass ffmpeg. 

Perhaps try using VLC to split the video into frames:
http://www.videolan.org/vlc/download-windows.html
https://www.isimonbrown.co.uk/vlc-export-frames/

If you able to split your video into frames then note down the directory where the split frames are placed. When you use the`trackPath` function from pathtrackR you will specify this directory as dirpath: 
`'trackPath = (dirpath, xarena, yarena, fps, box = 2, contrast = 0.5)'`

**Comment by Sylvain6259** [comment]
Hi dbthomas, thank for your answer.
Ok , I will try to split my video myself. And also try with ffmepg from another laptop.

I have just tried...
I may be stupide but how do you set the dirpath...
I have tried this
`> trackPath = (dirpath = "D:/Boulot/track2017/fra", xarena = 10, yarena = 10, fps = 30,contrast=0.5) 
`or
`> trackPath = ("D:/Boulot/track2017/fra", xarena = 10, yarena = 10, fps = 30,contrast=0.5)
`Everytime I got this error :

`Error: unexpected ',' in "trackPath = ("D:/Boulot/track2017/fra","
`

I would have a `,` that should not be here... I really feel stupid righ now, it seems to me that is a very simple trick.

**Comment by bpomalle** [comment]
I am running into an error using _trackPath_
I run the function, am prompted to draw a box around my animal, I click twice to draw the animal's box. Then, I get an error message before I can draw a box around the arena. This is the message:
`> trackPath('C:/AAM/Prestrike')
Select a portion of the image that includes the entire animal...
Error in as.vector(raster::extent(raster::select(raster::raster(file.list[1],  : 
  no method for coercing this S4 class to a vector`
And the directory I call to was created from splitVideo and only contains the jpegs. Any advice?

**Comment by dbthomas** [comment]
Hi bpomalle,
1) I will just leave the link to some visual instructions for pathtrackR here in case you or any other reader haven't seen this page: `https://aharmer.github.io/pathtrackr/`
2) Just to help me understand the issue - when you say you click twice to draw the animal's box, are you clicking on opposite corners of the animal to draw the bounding box? For example, in the tutorial link above there is an image of a spider with a red box around it. The box for the spider is defined by first clicking near the top left of the spider, and then by clicking near the bottom right of the spider. Is that what you have also been doing?
3) The purpose of the `splitVideo` function is to split the video into individual frames, which are stored as jpegs. A directory full of jpegs is the expected result.

**Comment by bpomalle** [comment]
Hi db, 

1. Thanks. Yes I have seen these visual instructions.
2. Yes. I am clicking on opposite corners to draw the animal's box. The red box then shows up around my animal. I'm assuming you then select opposite corners around the arena (e.g. the white tray your spider is in, for example) to define the arena. However, it does not let me do that.  
3. Understood. Just thought I would let you know that the splitVideo function worked fine and produced the desired result.

I can upload  mp4 of my video if that helps.

**Comment by dbthomas** [comment]
Sure thing, feel free to upload your video and aharmer and I will see how we can help.

**Comment by bpomalle** [comment]
Great! Here is a link to one of my videos:
[https://drive.google.com/open?id=0BzsCt97V7rI9M1I4VzN5d1BUS28](url)

**Comment by aharmer** [comment]
Hi bpomalle,

I was not able to reproduce your error message. As you can see [here](https://drive.google.com/file/d/0B_qAkhrxdpHwUE1DQzNYOFcwcnM/view?usp=sharing) I was able track your fish without issue (box size was reduced to 1).

My best guess is that there is a conflict with another package. You could try manually reloading the 'raster' package to make sure it takes precedence. 

Also, make sure you specify the arena dimensions and fps when you call `trackPath`. I noticed above you only specified the directory.

**Comment by bpomalle** [comment]
Thanks for your help aharmer and db. All the functions worked for me on a different computer after I installed the most recent version of R on that PC. I also had to install `rgdal`, which I now realize was not compatible with older versions of R. So, I think I just need to update R on my laptop then I should be good to go! I'm becoming a big fan of `pathtrackr`, thanks.
